### PR TITLE
fix: avoid falling back to random port if desired port not free

### PR DESCRIPTION
**Pull request by joske**
If you ask for a desired port and it's not available, the TCP stack should not silently choose a random port, but return an error.

Some tests needed to be fixed for this, as they were all trying to bind to the same port, making parallel test execution impossible.

### CODE REVIEW

File: node/src/beacon/mod.rs [hunk 1]
```diff
@@ -70,13 +70,15 @@ pub struct Beacon<N: Network, C: ConsensusStorage<N>> {
...
     /// Initializes a new beacon node.
+    #[allow(clippy::too_many_arguments)]
     pub async fn new(
         node_ip: SocketAddr,
         rest_ip: Option<SocketAddr>,
         account: Account<N>,
         trusted_peers: &[SocketAddr],
         genesis: Block<N>,
         cdn: Option<String>,
+        bft_ip: Option<SocketAddr>,
```
Comment (niklaslong) [inline node/src/beacon/mod.rs]: I'd maybe group it with the rest of the IPs?

File: node/tcp/src/helpers/config.rs [hunk 2]
```diff
@@ -84,7 +84,7 @@ impl Default for Config {
             name: None,
             listener_ip: default_ip(),
             desired_listening_port: None,
-            allow_random_port: true,
+            allow_random_port: false,
```
Comment (ljedrz) [inline node/tcp/src/helpers/config.rs:87]: this belongs in a separate commit IMO, together with the necessary test changes

Comment (joske) [reply node/tcp/src/helpers/config.rs:87]: Not sure what you mean? All the rest ARE the necessary test changes. I've split up this change from the test changes. Is that what you meant?

Comment (ljedrz) [reply node/tcp/src/helpers/config.rs:87]: sorry for not being clear; I meant the extension of the node creation methods, but I'm guessing those were necessary for the test adjustments

Comment (joske) [reply node/tcp/src/helpers/config.rs:87]: Indeed. Still it makes sense to split of the actual fix from the test fixes.

**Comment by howardwu** [comment]
I would request we take a different approach to handling this issue.

For instance, one could use serial_test or for the test cases add a cfg on a /tmp counter to help assist in this.

**Comment by joske** [comment]
The reason I opened this PR is a conceptual one: it seems wrong to fallback to a random port when you ask for a specific one. 

But, if we fix that, then the tests break, because they rely on the random port fallback.

I'm fine to leave this as is, if you think the fallback problem is not an issue.
